plugins/auth-backend/src/providers/oauth2/provider.ts

```diff
@@ -168,17 +173,20 @@ export const createOAuth2Provider: AuthProviderFactory = ({
     const callbackUrl = `${globalConfig.baseUrl}/${providerId}/handler/frame`;
     const authorizationUrl = envConfig.getString('authorizationUrl');
     const tokenUrl = envConfig.getString('tokenUrl');
+    const disableRefresh = envConfig.getBoolean('disableRefresh');
+    const customAttributes = envConfig.getStringArray('customAttributes');
 
     const provider = new OAuth2AuthProvider({
       clientId,
       clientSecret,
       callbackUrl,
       authorizationUrl,
       tokenUrl,
+      customAttributes,
     });
 
     return OAuthAdapter.fromConfig(globalConfig, provider, {
-      disableRefresh: false,
+      disableRefresh: disableRefresh,
```

Comment:
Do you absolutely need this right now or is it more of a nice thing to keep things clean? The auth-backend will be refactored slightly to provide this configuration through code instead.

Reply:
No, we don't need to right away.  I thought that it would be beneficial to users to specify if they want to disableRefresh instead of hard coding the value. Auth breaks if the auth used does not have a refresh token included.

If you can provide a way to toggle this option another way through code that would work. Maybe as a option when setting the providers in app package. But locking things behind packages that users consume as node modules may cause problems.